src/main/java/org/prebid/server/bidder/yieldlab/YieldlabBidder.java

```diff
@@ -231,6 +244,60 @@ private static String getConsentParameter(User user) {
         return ObjectUtils.defaultIfNull(consent, "");
     }
 
+    private static Map<String, String> extractDsaRequestParamsFromBidRequest(BidRequest request) {
+        return Optional.ofNullable(request.getRegs())
+            .map(Regs::getExt)
+            .map(ExtRegs::getDsa)
+            .map(YieldlabBidder::extractDsaRequestParamsFromDsaRegsExtension)
+            .orElseGet(Map::of);
+    }
+
+    private static Map<String, String> extractDsaRequestParamsFromDsaRegsExtension(final ExtRegsDsa dsa) {
+        final HashMap<String, String> dsaRequestParams = new HashMap<>();
+
+        if (dsa.getDsaRequired() != null) {
+            dsaRequestParams.put("dsarequired", dsa.getDsaRequired().toString());
+        }
+
+        if (dsa.getPubRender() != null) {
+            dsaRequestParams.put("dsapubrender", dsa.getPubRender().toString());
+        }
+
+        if (dsa.getDataToPub() != null) {
+            dsaRequestParams.put("dsadatatopub", dsa.getDataToPub().toString());
+        }
```

Comment:
I suggest going on with Optional since you started

`Optional.ofNullable(dsa.getRequired()).ifPresent(dsaRequired -> params.put(...));`

Reply:
This strikes me as an overuse of Optional.  Are you sure about this?

Reply:
Could you explain your concern regarding Optionals? 

It's just a way to handle possible NPEs with usually less code.

Reply:
To me this looks like [recipe 12](https://blogs.oracle.com/javamagazine/post/12-recipes-for-using-the-optional-class-as-its-meant-to-be-used). We know `dsa` is not null because that is checked further up, so all this would achieve is a different syntax for checking for null values at the expense of the additional memory footprint of instantiating an Optional. Granted, that cost is not high, but I don't see any benefit.

Reply:
Got you, the expenses of memory are hard to judge since we don't analyze how much Optionals hurt the system, that's why it's hard, at least for me, to claim what is the over-use and what is not.

Since the current version of the code doesn't go against the code style of the project and assuming you (or your team) are maintaining this adapter I'm fine with what you think is better option.